### PR TITLE
Add color-hex-alpha rule

### PR DESCRIPTION
**Pull request by jeddy3**
> Which issue, if any, is this issue related to?

Closes https://github.com/stylelint/stylelint/pull/4891
Closes https://github.com/stylelint/stylelint/issues/4880

> Is there anything in the PR that needs further explanation?

Implements the remaining requested changes in https://github.com/stylelint/stylelint/pull/4891.

Most of the tests were failing, so refactored to tidy things up and fixed the bugs.

### CODE REVIEW

File: lib/rules/color-hex-alpha/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# color-hex-alpha
+
+Require or disallow alpha channel for hex colors.
+
+<!-- prettier-ignore -->
+```css
+a { color: #ffff }
```
Comment (ybiquitous) [inline lib/rules/color-hex-alpha/README.md:7]: [suggestion] How about changing the alpha channel character for clarity and consistency with the following code examples? ⬇️ 

```suggestion
a { color: #fffa }
```

File: lib/rules/color-hex-alpha/__tests__/index.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+			message: messages.unexpected('#ffff'),
+			line: 1,
+			column: 12,
+		},
+	],
+});
+
+testRule({
+	ruleName,
+	config: ['always'],
```
Comment (ybiquitous) [inline lib/rules/color-hex-alpha/__tests__/index.js:70]: [trivial] How about the following change for consistency with other test cases?

```suggestion
	config: 'always',
```

File: lib/rules/color-hex-alpha/index.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	};
+}
+
+function isUrlFunction({ type, value }) {
+	return type === 'function' && value === 'url';
+}
+
+function isHexColor({ type, value }) {
+	return type === 'word' && HEX.test(value);
+}
```
Comment (ybiquitous) [inline lib/rules/color-hex-alpha/index.js:60]: [question] There are similar functions named `isHexColor` in other rules. Does it make sense to extract a utility function, like `lib/utils/isHexColor.js`? (maybe out of the scope of this PR)

https://github.com/stylelint/stylelint/blob/b01ed25dfa3e8231a976eef76bd5e81fb535b1e9/lib/rules/color-hex-case/index.js#L76-L78

https://github.com/stylelint/stylelint/blob/b01ed25dfa3e8231a976eef76bd5e81fb535b1e9/lib/rules/color-hex-length/index.js#L114-L116

**Review by ybiquitous — APPROVED** [review]
I've left some trivial comments, but it looks good to me. 👍🏼

File: lib/rules/color-hex-alpha/index.js (hunk 3 above)
Comment (jeddy3) [reply lib/rules/color-hex-alpha/index.js:60]: Good idea, but one for a refactor pull request.

We'll need to take into account that `color-hex-case` should check anything hex-like, whereas the other rules should only work on valid hex colours.

Comment (ybiquitous) [reply lib/rules/color-hex-alpha/index.js:60]: Thanks, I see that regexes are slightly different for each rule with reason.
It seems reasonable to keep them as-is.

**Comment by jeddy3** [comment]
> - Added: `color-hex-alpha` rule ([#5316](https://github.com/stylelint/stylelint/pull/5316)).

**Pull request merged**